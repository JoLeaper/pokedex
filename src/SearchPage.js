import React, { Component } from 'react'
import SearchResults from './SearchResults.js'
import SearchBar from './SearchBar.js'
import request from 'superagent';
import './SearchPage.css'

export default class App extends Component {
  state = {
    displayOrder: 'asc',
    pokemonName: '',
    pokemonType: [' ', 'normal', 'fire', 'fighting', 'water', 'flying', 'grass', 'poison', 'electric', 'ground', 'psychic', 'rock', 'ice', 'bug', 'dragon', 'ghost', 'dark', 'steel', 'fairy'],
    desiredPokemonType: '',
    pokemonAttack: 0,
    pokemon: [],
    page: 1,
    link: 'https://alchemy-pokedex.herokuapp.com/api/pokedex?'
  }
  componentDidMount = async () => {
    const fetchedPokemon = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex`);
    this.setState({ pokemon: fetchedPokemon.body.results })
}

  handleOrderChange = (event) => {
    this.setState({
      displayOrder: event.target.value
    })
  }

  handleTypeChange = (event) => {
    this.setState({
      desiredPokemonType: event.target.value
    })
  }

  handleNameChange = (event) => {
    this.setState({
      pokemonName: event.target.value
    })
  }

  handlePageChange = (event) => {
    let currentPage = this.state.page;
    if (event.target.value === 'next') {
      currentPage++;
    } else {
      currentPage--;
    }
    this.setState({
      page: currentPage
    })
  }

  handleAttackChange = (event) => {
      const re = /^[0-9\b]+$/;
      if (event.target.value === '' || re.test(event.target.value)) {
         this.setState({
          pokemonAttack: event.target.value
         })
      }
  }

  handleClick = async () => {
    let link = 'https://alchemy-pokedex.herokuapp.com/api/pokedex?' 
    const currentPage = 'page=' + this.state.page;
    const searchedPokemon = 'pokemon=' + this.state.pokemonName;
    const minAttack = 'attack=' + this.state.pokemonAttack;
    
    let wantedType;
    if (this.state.desiredPokemonType !== 'none') {
      wantedType='type=' + this.state.desiredPokemonType
    }
    link = `${link}&${currentPage}&${searchedPokemon}&${minAttack}&${wantedType}`
    console.log(link);
    const fetchedPokemon = await request.get(`${link}`)
    this.setState({ pokemon: fetchedPokemon.body.results })
  }

  handlePageClick = (event) => {
    const currentPage = this.state.page;
    if(event.target.value === 'next' && currentPage < this.state.pokemon.length) {
      this.setState(prevState => ({ page: prevState.page + 1 }))
    }

    if(event.target.value === 'prev' && currentPage > 1) {
      this.setState(prevState => ({ page: prevState.page - 1 }))
    }

    this.handleClick();
  }

  render() {
    return (
      <div className='search'>
        <SearchBar handlePageClick={this.handlePageClick} handleClick={this.handleClick} handleOrderChange={this.handleOrderChange} handleTypeChange={this.handleTypeChange} handleNameChange={this.handleNameChange} handleAttackChange={this.handleAttackChange} pokemonType={this.state.pokemonType} pokemonName={this.state.pokemonName} pokemonAttack={this.state.pokemonAttack} displayOrder={this.state.displayOrder}/>
        <SearchResults pokemonList={this.state.pokemon}/>
      </div>
    )
  }
}
