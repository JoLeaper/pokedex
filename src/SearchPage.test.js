import React from 'react';
import { shallow } from 'enzyme';
import SearchPage from './SearchPage'

test('renders learn react link', () => {
  const wrapper = shallow(<SearchPage />);
  expect(wrapper).toMatchSnapshot();
});



