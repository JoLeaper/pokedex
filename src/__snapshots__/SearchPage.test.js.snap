// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search page 1`] = `
<div
  className="search"
>
  <SearchBar
    displayOrder="asc"
    handleAttackChange={[Function]}
    handleClick={[Function]}
    handleNameChange={[Function]}
    handleOrderChange={[Function]}
    handlePageClick={[Function]}
    handleSortChange={[Function]}
    handleTypeChange={[Function]}
    pokemonAttack={0}
    pokemonName={null}
    pokemonType={
      Array [
        "",
        "normal",
        "fire",
        "fighting",
        "water",
        "flying",
        "grass",
        "poison",
        "electric",
        "ground",
        "psychic",
        "rock",
        "ice",
        "bug",
        "dragon",
        "ghost",
        "dark",
        "steel",
        "fairy",
      ]
    }
    sortBy="id"
  />
  <SearchResults
    pokemonList={Array []}
  />
</div>
`;
